### PR TITLE
fix: add type to test code

### PR DESCRIPTION
**Pull request by miettal**
```
$ npx tsc --strict
Found 3646 errors in 210 files.
↓
Found 1492 errors in 211 files.
```

### CODE REVIEW

File: ui/src/app/shared/utils/date/dateutils.spec.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ describe('DateUtils', () => {
   });
 
   it('#isDateBefore - checks if given date is before date to be compared to', () => {
-    const date: Date = DateUtils.stringToDate('2023-01-01');
```
Comment (miettal) [inline ui/src/app/shared/utils/date/dateutils.spec.ts:40]: input is gived by immediate value. so this code guarantee type is Date.

File: ui/src/app/shared/utils/array/array.utils.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export namespace ArrayUtils {
    * @param arr the arr
    * @returns a number if arr not empty, else null
    */
-  export function findBiggestNumber(arr: number[]): number | null {
```
Comment (miettal) [inline ui/src/app/shared/utils/array/array.utils.ts:26]: test pattern has null and undefined. so this function work with null and undefined.

File: ui/src/app/shared/genericComponents/shared/testing/tester.ts [hunk 3]
```diff
@@ -207,7 +207,7 @@ export namespace OeChartTester {
     export type Data = {
       type: 'data',
       label: string | Converter,
-      value: number[] | null
```
Comment (miettal) [inline ui/src/app/shared/genericComponents/shared/testing/tester.ts:210]: test data inlucde null element.

File: ui/src/app/shared/utils/date/dateutils.spec.ts [hunk 4]
```diff
@@ -37,7 +37,7 @@ describe('DateUtils', () => {
...
+    const date: Date = DateUtils.stringToDate('2023-01-01') as Date;
```
Comment (venu-sagar) [inline ui/src/app/shared/utils/date/dateutils.spec.ts:40]: This works. lines below it also follows similar pattern. a slight modification like below can solve 5 other strict errors.
isDateBefore(date: Date, compareDate: Date) -> isDateBefore(date: Date | null, compareDate: Date | null).

But it's upto you.

**Review by venu-sagar — APPROVED** [review]

File: ui/src/app/shared/utils/date/dateutils.spec.ts (hunk 4 above)
Comment (miettal) [reply ui/src/app/shared/utils/date/dateutils.spec.ts:40]: I see. thank you for your feedback. This time I want to focus just adding types, So I will keep this line. But I will memorize this information in mind.

**Comment by miettal** [comment]
rebased!

**Comment by sfeilmeier** [comment]
If I run:

`npm audit fix && npm install && node_modules/.bin/ng lint --fix; node_modules/.bin/tsc && node_modules/.bin/tsc-strict && node_modules/.bin/ng build -c "openems,openems-backend-prod,prod"`

I get

```
src/app/shared/utils/array/array.utils.ts(17,21): error TS2345: Argument of type 'number | null | undefined' is not assignable to parameter of type 'number'.
src/app/shared/utils/array/array.utils.ts(31,21): error TS2345: Argument of type 'number | null | undefined' is not assignable to parameter of type 'number'.
src/app/edge/live/Controller/Ess/TimeOfUseTariff/flat/flat.ts(15,15): error TS2564: Property 'priceWithCurrency' has no initializer and is not definitely assigned in the constructor.
```

**Comment by miettal** [comment]
Sorry, I mis-rebased. now its ok.

findSmallestNumber and findBiggestNumber was refactord.
https://github.com/OpenEMS/openems/pull/2702/commits/e228a2091656f8c642daca7d83a0038a0b6eb22a

**Review by sfeilmeier — APPROVED** [review]
Thank you! 👍

**Pull request merged**